### PR TITLE
improve graph loading time by noting multi-edge matrices

### PR DESCRIPTION
**Pull request by swilly22**

### CODE REVIEW

File: src/serializers/decoders/current/v8/decode_graph.c [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	// Load the key schema.
+	PayloadInfo *key_schema = _RdbLoadKeySchema(rdb);
+
+	/* The decode process contains the decode operation of many meta keys, representing independent parts of the graph.
+	 * Each key contains data on one or more of the following:
+	 * 1. Nodes - The nodes that are currently valid in the graph.
+	 * 2. Deleted nodes - Nodes that were deleted and there ids can be re-used. Used for exact replication of data black state.
+	 * 3. Edges - The edges that are currently valid in the graph.
+	 * 4. Deleted edges - Edges that were deleted and there ids can be re-used. Used for exact replication of data black state.
+	 * 5. Graph schema - Propertoes, indices.
```
Comment (jeffreylovitz) [inline src/serializers/decoders/current/v8/decode_graph.c:136]: Nit - "Properties'

File: src/serializers/decoders/current/v8/decode_graph.c [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+	 * */
+
+	GraphContext *gc = _DecodeHeader(rdb);
+	// Load the key schema.
+	PayloadInfo *key_schema = _RdbLoadKeySchema(rdb);
+
+	/* The decode process contains the decode operation of many meta keys, representing independent parts of the graph.
+	 * Each key contains data on one or more of the following:
+	 * 1. Nodes - The nodes that are currently valid in the graph.
+	 * 2. Deleted nodes - Nodes that were deleted and there ids can be re-used. Used for exact replication of data black state.
```
Comment (jeffreylovitz) [inline src/serializers/decoders/current/v8/decode_graph.c:133]: Nit - "data block state"

File: src/serializers/decoders/current/v8/decode_graph.c [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+	 * 3. Edges - The edges that are currently valid in the graph.
+	 * 4. Deleted edges - Edges that were deleted and there ids can be re-used. Used for exact replication of data black state.
```
Comment (jeffreylovitz) [inline src/serializers/decoders/current/v8/decode_graph.c:135]: Nit - "data block state"

**Review by jeffreylovitz — CHANGES_REQUESTED** [review]
This looks good to me provided that it has been thoroughly manually tested.

File: src/serializers/decoders/current/v8/decode_graph.c [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+static GraphContext *_DecodeHeader(RedisModuleIO *rdb) {
+	/* Header format:
+	 * Graph name
+	 * Node count
+	 * Edge count
+	 * Label matrix count
+	 * Relation matrix count - N
+	 * Does relationship Ri holds mutiple edges under a single entry X N
```
[the remark this answers is not in the document]
Comment (swilly22) [reply src/serializers/decoders/current/v8/decode_graph.c:53]: 👍

File: src/serializers/decoders/current/v8/decode_graph.c (hunk 1 above)
Comment (swilly22) [reply src/serializers/decoders/current/v8/decode_graph.c:136]: 👍

File: src/serializers/decoders/current/v8/decode_graph.c (hunk 2 above)
Comment (swilly22) [reply src/serializers/decoders/current/v8/decode_graph.c:133]: 👍

File: src/serializers/decoders/current/v8/decode_graph.c (hunk 3 above)
Comment (swilly22) [reply src/serializers/decoders/current/v8/decode_graph.c:135]: 👍

**Review by jeffreylovitz — APPROVED** [review]
The code looks good and I did some cursory testing of this locally, everything seems to behave as expected.

Nice addition!

**Pull request merged**